tests/src/tracing/eventpipe/eventsvalidation/GCStartStop.cs

```diff
@@ -0,0 +1,54 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics.Tracing;
+using System.Collections.Generic;
+using Microsoft.Diagnostics.Tools.RuntimeClient;
+using Microsoft.Diagnostics.Tracing;
+using Tracing.Tests.Common;
+
+namespace Tracing.Tests.GCStartStop
+{
+    public class ProviderValidation
+    {
+        public static int Main(string[] args)
+        {
+            var providers = new List<Provider>()
+            {
+                new Provider("Microsoft-DotNETCore-SampleProfiler"),
+                new Provider("Microsoft-Windows-DotNETRuntime", 0b1, EventLevel.Informational)
+            };
+            
+            var configuration = new SessionConfiguration(circularBufferSizeMB: 1024, format: EventPipeSerializationFormat.NetTrace,  providers: providers);
+            return IpcTraceTest.RunAndValidateEventCounts(_expectedEventCounts, _eventGeneratingAction, configuration, _DoesTraceContainEvents);
+        }
+
+        private static Dictionary<string, ExpectedEventCount> _expectedEventCounts = new Dictionary<string, ExpectedEventCount>()
+        {
+            { "Microsoft-Windows-DotNETRuntime", -1 },
+            { "Microsoft-Windows-DotNETRuntimeRundown", -1 },
+            { "Microsoft-DotNETCore-SampleProfiler", -1 }
+        };
+
+        private static Action _eventGeneratingAction = () => 
+        {
+            for (int i = 0; i < 1000; i++)
+            {
+                ProviderValidation providerValidation = new ProviderValidation();
+                providerValidation = null;
+                GC.Collect();
+            }
+        };
+
+        private static Func<EventPipeEventSource, Func<int>> _DoesTraceContainEvents = (source) => 
+        {
+            int GCStartEvents = 0;
+            int GCEndEvents = 0;
+            source.Clr.GCStart += (eventData) => GCStartEvents += 1;
+            source.Clr.GCStop += (eventData) => GCEndEvents += 1;
+            return () => GCStartEvents >= 1000 && GCEndEvents <= 1000 && GCStartEvents == GCEndEvents ? 100 : -1;
```

Comment:
Per @sywhang's comment, I think this is a good spot to put some logging. That being said, I think that these may be timeouts rather than test failures, despite there not being a "timeout" message.  The runtime looks to be just over 5 minutes, which I think is the time limit for tests.  Could you reduce the number of forced GCs to something lower like 200 and see if we still see this failure?

Reply:
Reduced the number of GCs to 100 and added some logging in case it still fails